### DISCUSSION TITLE
Is it time to switch to Vite.js?

### DISCUSSION THREAD

**Discussion by martinszeltins** [webpack/webpack#15931] [General]
A lot of web developers seem to be switching to Vite.js these days. It seems a lot faster (especially for larger projects) since it doesn't compile the whole app but only the parts that are needed. What are your thoughts?

**Comment by alexander-akait** [comment]
There is good article https://storybook.js.org/blog/storybook-performance-from-webpack-to-vite/#:~:text=Compared%20to%20Webpack%205%20with,your%20changes%20in%20the%20browser. and you can see another results

Notes for lc/cs - lc - https://webpack.js.org/configuration/experiments/#experimentslazycompilation and it speeds up your builds, layer cache - https://webpack.js.org/configuration/cache/#cachetype (more perf)

Webpack is very stable, especial for production builds.

And if you have perf problems with `babel` (this is the bottleneck for most projects), you can try https://github.com/swc-project/swc-loader and swc as minifier (compress better than esbuild, vite uses esbuild under the hood) - https://webpack.js.org/plugins/terser-webpack-plugin/#swc, using `swc` you will have even better results
